### ISSUE TITLE
Unable to install/upgrade to 2.6 inside of docker ubuntu18.04

### ISSUE THREAD

**Issue by dbilleci-lightstream** [pyca/cryptography#4791]
Gives error on 'futures' having a syntax error.  Reverting to 2.5 is a workaround.

````
    Step 21/32 : RUN pip3 install --upgrade cryptography
     ---> Running in e26374e1ed69
    Collecting cryptography
      Downloading https://files.pythonhosted.org/packages/a3/5f/d5b7d21006e3c1a3919a3cc14e2a5ce1cee1c7ff635f09b31d91bdaff377/cryptography-2.6.tar.gz (492kB)                                                                                                                       
      Installing build dependencies: started
      Installing build dependencies: finished with status 'error'
      Complete output from command /usr/bin/python3 /usr/local/lib/python3.6/dist-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-q21p97gl/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools>=18.5 wheel "cffi>=1.8,!=1.11.3; python_implementation != 'PyPy'":
      Traceback (most recent call last):
        File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
          "__main__", mod_spec)
        File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
          exec(code, run_globals)
        File "/usr/local/lib/python3.6/dist-packages/pip/__main__.py", line 16, in <module>
          from pip._internal import main as _main  # isort:skip # noqa
        File "/usr/local/lib/python3.6/dist-packages/pip/_internal/__init__.py", line 40, in <module>
          from pip._internal.cli.autocompletion import autocomplete
        File "/usr/local/lib/python3.6/dist-packages/pip/_internal/cli/autocompletion.py", line 8, in <module>
          from pip._internal.cli.main_parser import create_main_parser
        File "/usr/local/lib/python3.6/dist-packages/pip/_internal/cli/main_parser.py", line 12, in <module>
          from pip._internal.commands import (
        File "/usr/local/lib/python3.6/dist-packages/pip/_internal/commands/__init__.py", line 6, in <module>
          from pip._internal.commands.completion import CompletionCommand
        File "/usr/local/lib/python3.6/dist-packages/pip/_internal/commands/completion.py", line 6, in <module>
          from pip._internal.cli.base_command import Command
        File "/usr/local/lib/python3.6/dist-packages/pip/_internal/cli/base_command.py", line 25, in <module>
          from pip._internal.index import PackageFinder
        File "/usr/local/lib/python3.6/dist-packages/pip/_internal/index.py", line 41, in <module>
          from pip._internal.wheel import Wheel
        File "/usr/local/lib/python3.6/dist-packages/pip/_internal/wheel.py", line 7, in <module>
          import compileall
        File "/usr/lib/python3.6/compileall.py", line 20, in <module>
          from concurrent.futures import ProcessPoolExecutor
        File "/usr/local/lib/python3.6/dist-packages/concurrent/futures/__init__.py", line 8, in <module>
          from concurrent.futures._base import (FIRST_COMPLETED,
        File "/usr/local/lib/python3.6/dist-packages/concurrent/futures/_base.py", line 381
          raise exception_type, self._exception, self._traceback
                              ^
      SyntaxError: invalid syntax
````

**Comment by alex** [comment]
Can you give us a complete dockerfile with which to reproduce this (Doesn't have to be the exact one you're using, just one where we can see this ourselves)?

**Comment by dbilleci-lightstream** [comment]
Sure, working on that now.. stripping down my current one a bit to reproduce the issue.

**Comment by dbilleci-lightstream** [comment]
k this still gives me same error:

````
    FROM ubuntu:18.04
    LABEL maintainer="dan@dan.com"
    
    # Main OS config
    ENV DEBIAN_FRONTEND=noninteractive
    RUN echo "Etc/UTC" > /etc/timezone
    
    # Main apt configure
    RUN apt-get update && \
        apt-get install sudo python3-pip python3-dev build-essential \
        libyaml-dev libz-dev locales curl jq \
        git wget unzip libffi-dev libssl-dev -y
    
    # Lang config
    RUN echo "en_US.UTF-8 UTF-8" > /etc/locale.gen && locale-gen en_US.UTF-8
    ENV LC_ALL en_US.utf8
    
    # Node
    RUN curl -sL https://deb.nodesource.com/setup_10.x | bash
    RUN apt-get install --yes nodejs
    
    # Clean apt
    RUN rm -rf /var/lib/apt/lists/*
    
    # Pip update pip
    RUN pip3 install --upgrade setuptools pip
    
    # Install other pkgs
    RUN pip3 install awscli azure-cli ansible
    
    # Lock to cryptography to 2.5, 2.6 is broken (workaround - commented out)
    # RUN pip3 install --upgrade cryptography==2.5
    
    # Show broken
    RUN pip3 install --upgrade cryptography
    
    WORKDIR /data
    
    CMD /bin/bash
````

**Comment by alex** [comment]
I don't quite understand what's going on here. One thing that jumps out to me in running this is that you install a cryptography from apt, and then overwrite it with one via pip. That combination tends to lead to trouble. I suspect if you installed your pip packages into a virtualenv, instead of globally, that'd avoid this issue.

It's also possible you're being hit by #4789

**Comment by dbilleci-lightstream** [comment]
Ok.. I'm not sure where I'm installing an apt version of cryptography, I'm assuming it's in one of those packages as a requirement, but I don't know where.  Also, since this is a docker container I prefer to not involve a virtual environment.  Maybe that's against best practices, but it's been working fine for quite some time and I'm not able to refactor right now.  Either way, this is more for your information - I'm good with locking to 2.5 for now.
Thanks!

**Comment by alex** [comment]
In any event, I suspect this will be fixed with #4789, so I'm going to close this as a dupe of that. If, once we release 2.6.1, that doesn't solve your issue for any reason, please leave a comment and we'll reopen.

**Comment by dbilleci-lightstream** [comment]
Sounds good, thank you.
